src/aws_encryption_sdk/streaming_client.py

```diff
@@ -486,14 +488,25 @@ def _write_header(self):
 
     def _prep_non_framed(self):
         """Prepare the opening data for a non-framed message."""
+        try:
+            plaintext_length = self.stream_length
```

Comment:
Is a streaming client created per stream, or is it thread safe? Curious how instance variables work in these objects.

Reply:
Streaming client instances are not threadsafe. Instances should not be shared across threads.

Class attributes defined and modified in the class definition (we don't have any of those) are global across all instances of that class. Any attributes defined or modified on the instance are local to that instance.

Reply:
Gotcha sounds good.